src/scrobbler/listenbrainzscrobbler.cpp

```diff
@@ -569,6 +583,19 @@ void ListenBrainzScrobbler::Submit() {
       object_track_metadata.insert("release_name", QJsonValue::fromVariant(item->album_));
     }
 
+    QJsonObject object_additional_info;
+
+    object_additional_info.insert("duration_ms", item->duration_ / kNsecPerMsec);
+
+    if (item->track_ > 0) {
+      object_additional_info.insert("tracknumber", item->track_);
+    }
+
+    object_additional_info.insert("submission_client", QCoreApplication::applicationName());
+    object_additional_info.insert("submission_client_version", QCoreApplication::applicationVersion());
+
+    object_track_metadata.insert("additional_info", object_additional_info);
```

Comment:
In my own code I often put such code into immediately invoked function expressions, usually to make the object `const`. This is not used in C++ often, but maybe you will like it.
```suggestion
    QJsonObject object_additional_info = [&]() {
      QJsonObject object_additional_info;

      object_additional_info.insert("duration_ms", item->duration_ / kNsecPerMsec);

      if (item->track_ > 0) {
        object_additional_info.insert("tracknumber", item->track_);
      }

      object_additional_info.insert("submission_client", QCoreApplication::applicationName());
      object_additional_info.insert("submission_client_version", QCoreApplication::applicationVersion());

      return object_additional_info;
    }();
```